src/plugins/data_source/server/client/data_source_client.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Logger, OpenSearchClient, SavedObjectsClientContract } from 'src/core/server';
+
+/**
+ * Represents an OpenSearch cluster API client created by the platform.
+ * It allows to call API on behalf of user defined in "data source" saved object
+ *
+ * @public
+ **/
+export interface IDataSourceClient {
+  /**
+   * Creates a {@link OpenSearchClient } bound to given data source
+   */
+  asDataSource: (dataSourceId: string) => Promise<OpenSearchClient>;
+}
+
+/**
+ * See {@link IDataSourceClient}
+ *
+ * @public
+ */
+export interface ICustomDataSourceClient extends IDataSourceClient {
+  /**
+   * Closes the data source client. After that client cannot be used and one should
+   * create a new client instance to be able to interact with OpenSearch API.
+   */
+  close: () => Promise<void>;
```

Comment:
why need a separate interface to include close method but not in IDataSourceClient?

Reply:
The `close()` function is a critical one that closes all the connections in pool, that we don't want to expose this function to data source client consumers, who only needs to retrieve the client

Reply:
close() should only be called when OSD or plugin is shut down

Reply:
where is this `close()` function being used?

Reply:
I forget to remove this. So after adding `DataSourceService` , we'll handle closing connections at the service level. Thanks for pointing it out